ash-bergs/dish-dollar
# [backend] feat: setup Vitest

**Description**:

This PR sets up `vitest` as our testing framework for the backend project and introduces unit tests for the User model. 

### Main Changes:

1. **Setup `vitest` in Backend Project**:
   - Integrated [vitest](https://vitest.dev/) into development workflow.
   - Configured necessary settings to ensure smooth running of tests.

2. **Prisma Client Mocking**:
   - Created a mock of the Prisma client. This allows us to write unit tests without the need for a real database connection.
   - This mock is essential for isolating the functions we want to test and ensuring our tests run quickly.

3. **User Model Unit Tests**:
   - **Create User Test**: Ensures that we can successfully create a user.
   - **Find User by ID Test**: Checks that our function retrieves the correct user based on a provided ID.
   - **Update User Information Test**: Verifies that we can update a user's information and that the updated data is reflected correctly.

### Useful Resources:
- Inspired by the [Prisma's Testing Series](https://www.prisma.io/blog/testing-series-1-8eRB5p0Y8o).

### Next Steps:
- Extend unit tests to other models.
- Explore integration tests for more complex workflows.
- Setup CI to run tests 

<img width="505" alt="image" src="https://github.com/ash-bergs/dish-dollar/assets/65979049/c008f65e-a1dd-4714-8006-4012e5c1b42e">
